contracts/test/psm/PegStabilityModuleNonCustodial.t.sol

```diff
@@ -0,0 +1,235 @@
+import {ICore} from "../../core/ICore.sol";
+import {Core} from "../../core/Core.sol";
+import {IFei, Fei} from "../../fei/Fei.sol";
+import {PegStabilityModuleNonCustodial} from "./../../peg/PegStabilityModuleNonCustodial.sol";
+import {PegStabilityModule} from "./../../peg/PegStabilityModule.sol";
+import {Vm} from "./../utils/Vm.sol";
+import {DSTest} from "./../utils/DSTest.sol";
+import {getCore, getAddresses, FeiTestAddresses} from "./../utils/Fixtures.sol";
+
+contract PegStabilityModuleNonCustodialTest is DSTest {
+
+    PegStabilityModuleNonCustodial private psm;
+    ICore private core;
+    IFei private fei;
+
+    modifier prank(address toPrank) {
+      vm.startPrank(toPrank);
+
+      _;
+
+      vm.stopPrank();
+    }
+
+    uint256 immutable public mintAmount = 10_000_000e18;
+    uint256 immutable public bufferCap = 10_000_000e18;
+    uint256 immutable public rps = 10_000e18;
+
+    uint256 immutable public floorBasisPoints = 9_900;
+    uint256 immutable public ceilingBasisPoints = 10_100;
+
+    MockERC20 public underlyingToken;
+    MockPCVDepositV2 public pcvDeposit;
+    MockOracle public oracle;
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+    FeiTestAddresses public addresses = getAddresses();
+    address immutable public userAddress = addresses.userAddress;
+
+    function setUp() public {
+      core = getCore();
+
+      fei = core.fei();
+
+      oracle = new MockOracle(1);
+      underlyingToken = new MockERC20();
+      pcvDeposit = new MockPCVDepositV2(
+          address(core),
+          address(underlyingToken),
+          0,
+          0
+      );
+
+      PegStabilityModule.OracleParams memory oracleParams = PegStabilityModule.OracleParams({
+          coreAddress: address(core),
+          oracleAddress: address(oracle),
+          backupOracle: address(0),
+          decimalsNormalizer: 0,
+          doInvert: false
+      });
+
+      psm = new PegStabilityModuleNonCustodial(
+        floorBasisPoints,
+        ceilingBasisPoints,
+        oracleParams,
+        0,
+        0,
+        rps,
+        bufferCap,
+        IERC20(underlyingToken),
+        IPCVDeposit(pcvDeposit)
+      );
+
+      underlyingToken.mint(address(psm), mintAmount);
+      underlyingToken.mint(userAddress, mintAmount);
+
+      vm.startPrank(addresses.governorAddress);
+
+      core.grantMinter(addresses.governorAddress);
+      core.grantPCVController(address(psm));
+      core.grantMinter(address(psm));
+      fei.mint(userAddress, mintAmount);
+
+      vm.stopPrank();
+
+      psm.allocateSurplus();
+    }
+
+    /// @notice PSM is set up correctly, all state variables and balances are correct 
+    function testPSMSetup() public {
+      uint256 startingPSMUnderlyingBalance = psm.balance();
+      uint256 startingUserFEIBalance = fei.balanceOf(userAddress);
+      uint256 reserveThreshold = psm.reservesThreshold();
+
+      assertEq(reserveThreshold, 0);
+      assertEq(startingPSMUnderlyingBalance, 0);
```

Comment:
Nice! Really like the idea of adding a natspace comment to make the test easier to understand